etl/steps/data/garden/animal_welfare/2023-08-03/share_of_cage_free_eggs.py

```diff
@@ -1,14 +1,41 @@
 """Load a meadow dataset and create a garden dataset."""
 
 import owid.catalog.processing as pr
-from owid.catalog import Dataset
+from owid.catalog import Dataset, Table
 
 from etl.helpers import PathFinder, create_dataset
 
 # Get paths and naming conventions for current step.
 paths = PathFinder(__file__)
 
 
+def add_share_of_eggs_in_uk(tb_uk: Table) -> Table:
+    tb_uk = tb_uk.copy()
+
+    # Add the share of UK eggs in cages and cage-free.
+    tb_uk["share_of_eggs_in_cages"] = 100 * tb_uk["number_of_eggs_from_enriched_cages"] / tb_uk["number_of_eggs_all"]
+    tb_uk["share_of_eggs_cage_free"] = 100 - tb_uk["share_of_eggs_in_cages"]
+
+    # Update their units.
+    tb_uk["share_of_eggs_cage_free"].metadata.unit = "%"
+    tb_uk["share_of_eggs_cage_free"].metadata.short_unit = "%"
+    tb_uk["share_of_eggs_in_cages"].metadata.unit = "%"
+    tb_uk["share_of_eggs_in_cages"].metadata.short_unit = "%"
```

Comment:
in case this were to grow...

```python
columns = ["share_of_eggs_cage_free", "share_of_eggs_in_cages"]
for column in columns:
  tb_uk[column].metadata.unit = "%"
  tb_uk[column].metadata.short_unit = "%"

Reply:
Same number of lines, not worth it :-). But one thing to keep in mind is that this issue may happen more often in the future: You construct a "share" variable, and need to change the units accordingly. Maybe we could create a simple function to create "share" variables (that would update the units automatically).